### DISCUSSION TITLE
Atom relational property name: rel_op vs. rop

### DISCUSSION THREAD

**Discussion by bertiewooster** [sympy/sympy#22603] [Ideas]
As @smichr noted on [StackOverflow](https://stackoverflow.com/questions/67153048/reliable-way-to-extract-values-from-parse-sympy-result), rel_op (<, <=, >, >=) is a property of an atom. However, the [atom API documentation page](https://docs.sympy.org/dev/modules/core.html?highlight=atoms%20relational#module-sympy.core.relational) does not mention rel_op, instead listing rop (along with lhs and rhs). In the documentation, should rop be changed to rel_op?

**Comment by oscargus** [comment]
It probably would make sense to do that in some ways. However, there is a possible issue if some downstream code uses `rop` as a keyword argument (it doesn't appear to be used anywhere in the SymPy base as a keyword argument). Also, there isn't a one-to-one mapping between `rop` and `rel_op` (e.g. `rop='<>'` works, but will return `'!='`).

Personally, I do not think it gives enough benefit to change it considering the potential negative consequences elsewhere.

It should be useful to mention that `rel_op` can be used to obtain the type in the documentation though. As well as the valid values for `rop`, maybe as a single table?

**Reply by bertiewooster** [reply]
Thanks for clarifying that `rel_op` and `rop` are both used.

I am unable to find rel_op anywhere in the documentation. So `rel_op` should also be discussed [under Relational on the atom API documentation page](https://docs.sympy.org/dev/modules/core.html?highlight=atoms%20relational#module-sympy.core.relational)?

Good idea to have a table of the valid values for `rop`: the documentation says "Valid values can be found in the keys of Relational.ValidRelationOperator," and I can't find "ValidRelationOperator" anywhere else in the documentation.

**Reply by oscargus** [reply]
I also noted the Relational.ValidRelationOperator-part. It basically directs the reader to the source, but not really straightforward to get to that from the docs...

**Reply by bertiewooster** [reply]
Indeed, and telling learners to go read the source code is not ideal...

**Comment by oscarbenjamin** [comment]
The `rel_op` attribute is for Relational rather than Atom. The `atoms` method is poorly named when used like `.atoms(Relational)` which actually returns non-atoms like `Gt(x, y)` etc.

**Reply by bertiewooster** [reply]
Got it--I poorly labeled that [link, it is in fact to the Relational object](https://docs.sympy.org/dev/modules/core.html?highlight=atoms%20relational#module-sympy.core.relational).

> The `atoms` method is poorly named when used like `.atoms(Relational)` which actually returns non-atoms like `Gt(x, y)` etc.

Yes, I was struggling with why for example `x < 7/10` is an atom when writing a [guide to processing solutions](https://github.com/sympy/sympy/wiki/Guide:-Parsing-solutions). Maybe I'll add a footnote noting its name is misleading.
